annotations/static/annotations/js/annotators/textdisplay.js

```diff
@@ -227,49 +279,141 @@ TextDisplay = {
+                                const offsets = updatedAppellation.position.position_value.split(',');
+                                
+                                /*
+                                 * We need to maintain two sets of position values for different purposes:
+                                 * 
+                                 * 1. position.startOffset and position.endOffset:
+                                 *    - Used by frontend components for visual rendering, this set is responsible for the highlight on screen
+                                 *    - Part of the position object used by getTextPosition() in util.js
+                                 *    - These determine where the highlight appears on screen
+                                 */
+                                updatedAppellation.position.startOffset = parseInt(offsets[0]);
+                                updatedAppellation.position.endOffset = parseInt(offsets[1]);
+                                
+                                /*
+                                 * 2. startPos and endPos:
+                                 *    - These are old fields from the database model
+                                 *    - Still used by some parts of the frontend code
+                                 *    - Need to be kept in sync for backwards compatibility (such as submitting quadruples etc)
+                                 *    - Direct properties on the appellation object, not in position
+                                 */
+                                updatedAppellation.startPos = parseInt(offsets[0]);
+                                updatedAppellation.endPos = parseInt(offsets[1]);
+                                
+                                // Clear editing state
+                                localStorage.removeItem('editingAppellation');
+                                self.isEditing = false;
+                                EventBus.$emit('cancelEdit');
+
+                                // Update text selection
+                                self.$root.$emit('appellationUpdated', updatedAppellation);
+                                
+                                self.resetTextSelection();
+                                
+                                // Show success message
+                                EventBus.$emit('showMessage', {
+                                    text: 'Successfully updated annotation',
+                                    type: 'success'
+                                });
+                            }).catch(error => {
+                                console.error('Failed to update appellation:', error);
+                                EventBus.$emit('showMessage', {
+                                    text: 'Failed to update annotation. Please try again.',
+                                    type: 'error'
+                                });
```

Comment:
shouldn't editing be canceled in this case as well (especially if the mouse selection is cleared afterwards)?